python/graphscope/analytical/app/java_app.py

```diff
@@ -0,0 +1,381 @@
+    )
+    out, err = parse_user_app_process.communicate()
+    logger.info(err)
+    for line in out.split("\n"):
+        logger.info(line)
+        if len(line) == 0:
+            continue
+        if line.find("PropertyDefaultApp") != -1:
+            _java_app_type = "property"
+            continue
+        if line.find("ProjectedDefaultApp") != -1:
+            _java_app_type = "projected"
+            continue
+        if line.find("Error") != -1:
+            raise Exception("Error occured in verifying user app")
+        if line.find("TypeParams") != -1:
+            _frag_param_str = line.split(":")[1].strip()
+            continue
+        if line.find("ContextType") != -1:
+            _java_inner_context_type = line.split(":")[1].strip()
+            continue
+    logger.info(
+        "Java app type: {}, frag type str: {}, ctx type: {}".format(
+            _java_app_type, _frag_param_str, _java_inner_context_type
+        )
+    )
+
+    # java_codegen_stderr_watcher = PipeWatcher(parse_user_app_process.stderr, sys.stdout)
+    # setattr(parse_user_app_process, "stderr_watcher", java_codegen_stderr_watcher)
+    parse_user_app_process.wait()
+    return _java_app_type, _frag_param_str, _java_inner_context_type
+
+
+def _type_param_consistent(graph_actucal_type_param, java_app_type_param):
+    if java_app_type_param == "java.lang.Long":
+        if graph_actucal_type_param in {"uint64_t", "int64_t"}:
+            return True
+        return False
+    if java_app_type_param == "java.lang.Double":
+        if graph_actucal_type_param in {"double"}:
+            return True
+        return False
+    if java_app_type_param == "java.lang.Integer":
+        if graph_actucal_type_param in {"int32_t", "uint32_t"}:
+            return True
+        return False
+    return False
+
+
+# Java codegen cp can be empty when no bitcode is generated.
+# def _construct_jvm_options_from_params(
+#     app_lib_dir, jar_unpacked_path, llvm4jni_output_dir, java_codegen_cp
+# ):
+#     performance_args = " "
+#     # May be we should move this out of here.
+#     if os.path.isfile(DEFAULT_JVM_CONFIG_FILE):
+#         with open(DEFAULT_JVM_CONFIG_FILE, "r") as stream:
+#             config = yaml.safe_load(stream)
+#             performance_args = config["jvm"]["performance"]
+#     else:
+#         logger.info("No jvm config found, we still proceed...")
+
+#     library_path = "-Djava.library.path={}".format(GRAPHSCOPE_JNI_LIB)
+#     runtime_class_path = "-Djava.class.path={}".format(GRAPHSCOPE_RUNTIME_JAR)
+#     # Put jni lib in user_class_path, so url class loader can find.
+#     user_class_path = "{}:{}:{}:{}:{}".format(
+#         app_lib_dir,
+#         GRAPHSCOPE_JNI_LIB,
+#         llvm4jni_output_dir,
+#         java_codegen_cp,
+#         jar_unpacked_path,
+#     )
+#     return (
+#         " ".join([performance_args, library_path, runtime_class_path, "-Xrs"]).strip(),
+#         user_class_path,
+#     )
+
+
+class JavaApp(AppAssets):
```

Comment:
`AppAssets` already support `cpp_pie` `cython_pregel` and `cython_pie`, so maybe it's better to enrich the the `AppAssets  class` to support java app rather than a new `class JavaApp`

Reply:
There are some dirty work such as parse Java app class, compatibility check between java app type and graph type. Adding these stuff to ```AppAssets``` will make it heavy and not generic, from my point.

Reply:
@sighingnow Any comments?

If `JavaApp` is needed, rename to `JavaAppAssets`